cumulusci/core/dependencies/dependencies.py

```diff
@@ -523,12 +530,65 @@ def install(self, context: BaseProjectConfig, org: OrgConfig):
             "namespace_strip": self.namespace_strip,
         }
 
-        package_zip = MetadataPackageZipBuilder.from_zipfile(
-            zip_src, options=options, logger=logger
-        ).as_base64()
+        # We have a zip file. Now, determine how to handle
+        # MDAPI/SFDX format, with or without a subfolder specified.
+
+        # In only two cases do we need to do a zip subset.
+        # Either we have a repo root in MDAPI format and need to get `src`,
+        # or we're deploying a subfolder that is in MDAPI format.
+        zip_extract_subfolder = None
+        if (
+            not self.subfolder
+            and get_source_format_for_zipfile(zip_src, "src") is SourceFormat.MDAPI
+        ):
+            zip_extract_subfolder = "src"
+        elif (
+            self.subfolder
+            and get_source_format_for_zipfile(zip_src, self.subfolder)
+            is SourceFormat.MDAPI
+        ):
+            zip_extract_subfolder = self.subfolder
```

Comment:
Could this be simplified with Extract Variable? For example:
```suggestion
        zip_extract_subfolder = self.subfolder or "src"
        source_format = get_source_format_for_zipfile(zip_src, zip_extract_subfolder)
        if source_format is not SourceFormat.MDAPI:
            zip_extract_subfolder = None
```